core/src/main/java/org/springframework/security/access/expression/method/ExpressionBasedAnnotationAttributeFactory.java

```diff
@@ -34,13 +37,24 @@
 		PrePostInvocationAttributeFactory {
 	private final Object parserLock = new Object();
 	private ExpressionParser parser;
-	private MethodSecurityExpressionHandler handler;
+	private SecurityExpressionHandler<MethodInvocation> handler;
 
 	public ExpressionBasedAnnotationAttributeFactory(
 			MethodSecurityExpressionHandler handler) {
 		this.handler = handler;
 	}
 
+	/**
+	 * Support for reactive method secutiry
+	 * @param handler The {@link ReactiveMethodSecurityExpressionHandler}
+	 * @since 5.1.2
+	 */
+	public ExpressionBasedAnnotationAttributeFactory(
```

Comment:
We cannot add a reactive interface to an implementation that is used in imperative world because we will get NoClassDefFoundErrors